core/src/main/java/com/scalar/db/transaction/consensuscommit/RecoveryHandler.java

```diff
@@ -52,17 +53,34 @@ public void recover(
 
   @VisibleForTesting
   void rollbackRecord(Selection selection, TransactionResult result) throws ExecutionException {
-    logger.debug(
-        "Rollback for {}, {} mutated by {}",
-        selection.getPartitionKey(),
-        selection.getClusteringKey(),
+    assert selection.forFullTableName().isPresent();
+
+    TransactionTableMetadata tableMetadata =
+        getTransactionTableMetadata(tableMetadataManager, selection);
+    Key partitionKey = ScalarDbUtils.getPartitionKey(result, tableMetadata.getTableMetadata());
+    Optional<Key> clusteringKey =
+        ScalarDbUtils.getClusteringKey(result, tableMetadata.getTableMetadata());
+
+    logger.info(
```

Comment:
Updated log level from `debug` to `info` in `RecoveryHandler`, as lazy recovery has been performed in background threads since version 3.16.